contracts/src/RiscZeroVerifierFallbackRouter.sol

```diff
@@ -0,0 +1,78 @@
+// Copyright 2025 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+pragma solidity ^0.8.9;
+
+import {IRiscZeroVerifier} from "./IRiscZeroVerifier.sol";
+import {RiscZeroVerifierRouter} from "./RiscZeroVerifierRouter.sol";
+
+/// @notice Router for IRiscZeroVerifier, allowing multiple implementations to be accessible behind a single address
+///         and a fallback to the canonical router.
+/// @dev Extends RiscZeroVerifierRouter to add fallback behavior.
+contract RiscZeroVerifierFallbackRouter is RiscZeroVerifierRouter {
+    /// @notice The canonical RISC Zero verifier router used as fallback.
+    IRiscZeroVerifier public immutable fallbackRouter;
```

Comment:
To function properly, this needs to be `RiscZeroVerifierRouter`. Instead of casting it in `addVerifier`, we could simply change the type here.
```suggestion
    RiscZeroVerifierRouter public immutable fallbackRouter;
```